src/Controllers/ElementalAreaController.php

```diff
@@ -45,132 +47,158 @@ public function getClientConfig()
         $clientConfig = parent::getClientConfig();
         $clientConfig['form']['elementForm'] = [
             'schemaUrl' => $this->Link('schema/elementForm'),
-            'saveUrl' => $this->Link('api/saveForm'),
-            'saveMethod' => 'post',
-            'payloadFormat' => 'json',
-            'formNameTemplate' => sprintf(static::FORM_NAME_TEMPLATE ?? '', '{id}'),
+            'formNameTemplate' => sprintf(static::FORM_NAME_TEMPLATE, '{id}'),
         ];
-
         // Configuration that is available per element type
         $clientConfig['elementTypes'] = ElementTypeRegistry::generate()->getDefinitions();
-
         return $clientConfig;
     }
 
     /**
-     * @param HTTPRequest|null $request
+     * Used for both:
+     * - GET requests to get the FormSchema via getElementForm() called from LeftAndMain::schema()
+     * - POST Requests to save the Form. Will be handled by to FormRequestHandler::httpSubmission()
+     * /admin/linkfield/elementForm/<ElementID>
+     *
+     * @param HTTPRequest $request - deprecated 5.3.0 Will be removed without equivalent functionality to replace it
      * @return Form
-     * @throws HTTPResponse_Exception
      */
     public function elementForm(HTTPRequest $request = null)
     {
-        // Get ID either from posted back value, or url parameter
-        if (!$request) {
-            $this->jsonError(400);
-            return null;
-        }
-        $id = $request->param('ID');
+        $request = $request ?: $this->getRequest();
+        $id = $request->param('ItemID');
+
+        // Respect previous behaviour
+        // This should just be removed later and so that it 404's below when BaseElement::get()->byID($id) fails
```

Comment:
404 is correct here because ItemID is in the URL. If it was post data then I agree that 400 is correct.

Reply:
If `$id` is **always**  in the URL and there's no way to get here without one, the condition here is misleading at best, since the only time it's hit is when `$id` is `0` explicitly, not when it's missing which is what the condition implies.

If that's all correct, please modify the condition to test explicitly for `$id === 0` and either return a 400 (because an ID of 0 is always going to be bad data) or a 404 (because I feel like you won't agree about the 400 :p)

Reply:
Changed to `$id === 0`